### DISCUSSION TITLE
Using a patch with a workspace that is set up with shared-workspace-lockfile set to `false`

### DISCUSSION THREAD

**Discussion by stevejay** [pnpm/pnpm#6821] [Q&A] [answered]
I have inherited a monorepository that was set up as a pnpm workspace. The flag `shared-workspace-lockfile` was set to `false` in the workspace configuration, so each package in the workspace has its own lockfile.

I need to apply a patch to a dependency used by one of the packages.  I can generate the patch file without issue -- the patch appears in the `patches` directory in the root directory and the `patchedDependencies` information gets added to the root `package.json` file. However, when I try to install packages (whether or not with `--fix-lockfile`), I get the following error:

```
 ERR_PNPM_PATCH_NOT_APPLIED  The following patches were not applied: next-auth@4.22.1

Either remove them from "patchedDependencies" or update them to match packages in your dependencies.
```

I have that exact version installed in the package in question. 

My suspicion is that it is due to the `shared-workspace-lockfile` flag being set to `false`. Is this possible? I tried patching a dependency in a [simplified workspace example](https://github.com/shawnmcknight/pnpm-patch-tests). It worked fine when using a shared workspace lockfile. But when I updated the example to not use a shared workspace lockfile and then applied the patch, I got same error message as above.

**Comment by await-ovo** [comment, marked as answer]
Now we can apply patches with `shared-workspace-lockfile=false` after https://github.com/pnpm/pnpm/pull/6889 ~
